### PR TITLE
Add support for Leap 15.1 to spacewalk-common-channels, mgr-create-bootstrap-repo and bootstrapping

### PR DESCRIPTION
**Pull request by juliogonzalez**
## What does this PR change?

Add support for openSUSE Leap 15.1 to spacewalk-common-channels and mgr-create-bootstrap-repo

## GUI diff

No difference.

- [x] **DONE**

## Documentation

https://github.com/SUSE/doc-susemanager/pull/454

- [x] **DONE**

## Test coverage
- No tests: Not covered by tests.

- [x] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/7777

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test" 
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"		 
- [ ] Re-run test "java_pgsql_tests"		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"

### CODE REVIEW

**Comment by mcalmer** [comment]
We could change also mgr-create-bootstrap-repo in this PR if you want?

**Comment by juliogonzalez** [comment]
@mcalmer, ACK.

Can you review the PR for the doc as well?

**Comment by mcalmer** [comment]
Found another place we need to change for Leap15.1:

    susemanager-utils/susemanager-sls/salt/certs/

There is a link `Leap15_0.sls -> SLES12.sls` . I think we need the same for Leap15_1.sls to SLES12.sls .

**Comment by juliogonzalez** [comment]
I am testing now, just in case.

If there are no problems, I will merge.

**Comment by juliogonzalez** [comment]
- spacewalk-common-channels: works
- reposyncing for all repos: works
- bootstrap repo creating: works
- bootstrapping Leap 15.0 traditional: testing (I am having problems creating a VM)

**Comment by juliogonzalez** [comment]
https://bugzilla.opensuse.org/show_bug.cgi?id=1134758

**Comment by juliogonzalez** [comment]
I was able to bootstrap, and get the system added to the server.

However I am worried about this error:

REGISTRATION
------------
* registering

*** this system should now be registered, please verify ***

```
* setting permissions to allow configuration management
  NOTE: use an activation key to subscribe to the tools
        channel and zypper install/update rhncfg-actions
Error setting permissions for configuration management.
    Please ensure that the activation key subscribes the
    system to the tools channel and zypper install/update rhncfg-actions.
```
The client was subscribed with an activation key that includes mgr-cfg-actions.

Something tells me that the script provoking this is calling rhncfg-ations directly, and not using the `provides`, and it should allow either options (for old and new client tools).

**Comment by juliogonzalez** [comment]
Mmmmm, or having a look at `spacewalk/certs-tools/rhn_bootstrap_strings.py`, maybe the problem is that `zypper refresh` does not run after the repos are added.

**Comment by juliogonzalez** [comment]
Mmmm... it seems the bootstrap script does NOT install rhncfg-actions or mgr-cfg-actions, so this could be a normal error.

@mcalmer any idea?

**Comment by mcalmer** [comment]
The bootstrap script does not install these tools. If functionality is needed before registration, the "permissions" are faked by touching the capabilities file directly.

After the registration, the fake file is removed and the expectation is, that the tool was installed during the registration with the activation-key.

If this was not the case you see this message.

- Was the package for config management part of the package list in the activation key?
- if yes, was it the new name or the old name?
- if it was the old name, try to find the error or try to use the new name.
- Does it work with the new name?

**Comment by juliogonzalez** [comment]
@mcalmer I just had the same idea, and was talking to QA about it.

No, the package was not part of the activation key.

Let me try again. I will test with both package names.

**Comment by juliogonzalez** [comment]
@mcalmer at the same time, I think the message could add a hint about that way of solving the problem:

First problem: what's the point of having the note? If things work, it's not need. If things don't work, we have the error:
>  NOTE: use an activation key to subscribe to the tools
>        channel and zypper install/update rhncfg-actions

Second problem:
>Error setting permissions for configuration management.
>    Please ensure that the activation key subscribes the
>    system to the tools channel and zypper install/update rhncfg-actions

For me this doesn't mean the package `rhncfg-actions` should be added to the activation key, but:
1. Ensure the activation key has the tools channel.
2. Install the package manually.

Should we mention:
1. Ensure the activation key has the tools channel, and rhncfg-actions (mgr-cfg-actions for newer client tools)
2. Install the package with zypper install/update rhncfg-actions (zypper install/update mgr-cfg-actions for newer client tools).

Opinions? @lkotek, since you opened a bug about it, what' do you think?

**Comment by lkotek** [comment]
> * Ensure the activation key has the tools channel, and rhncfg-actions (mgr-cfg-actions for newer client tools)
> 
> * Install the package with zypper install/update rhncfg-actions (zypper install/update mgr-cfg-actions for newer client tools).

@juliogonzalez I think both solutions you suggest are valid. I think mentioning the first one ("Ensure the activation key has the tools channel") is quite clear and sufficient.

**Comment by juliogonzalez** [comment]
OK, let's do it on a separate PR.

**Comment by juliogonzalez** [comment]
> Was the package for config management part of the package list in the activation key?

Now it  is

> if yes, was it the new name or the old name?

Old name. Doesn't work.

> if it was the old name, try to find the error or try to use the new name.

Same error as when no package was added. Trying with new name.

> Does it work with the new name?

Doesn't work neither, but I wonder if this is related to https://github.com/uyuni-project/uyuni/issues/952 (basically Jabberd is dead).

**Comment by juliogonzalez** [comment]
GPG key was missing. Trying again tomorrow.

After that the bootstrap script generator needs to be fixed to make mention to the new package names (mgr-cfg*)

**Comment by juliogonzalez** [comment]
Tested and working. Fixing conflicts and mergng.
